src/test/java/org/isf/opd/test/TestOpd.java

```diff
@@ -84,8 +85,12 @@ public void _setParameters(Patient patient, Disease disease, Opd opd) {
 
 	public void check(Opd opd) {
 		assertThat(opd.getVisitDate()).isEqualTo(visitDate);
-		assertThat(opd.getAge()).isEqualTo(age);
-		assertThat(opd.getSex()).isEqualTo(sex);
+		if (!(GeneralData.OPDEXTENDED && opd.getPatient() != null)) {
+			// skip checks as OpdBrowserManager sets values from patient
+			// thus only do checks when not OPDEXTENDED and patient == null
+			assertThat(opd.getAge()).isEqualTo(age);
+			assertThat(opd.getSex()).isEqualTo(sex);
+		}
```

Comment:
same logic as in Laboratory, please read my comment [here](https://github.com/informatici/openhospital-core/pull/324#discussion_r575049087)

Reply:
@mwithi I need to leave this check here (and in the other repo) because on insert the patient's age is copied from the patient's record and doesn't match the value in `opd`.

Remember that in the Patient record there is this code:
````
private static Date birthDate = new GregorianCalendar(1984, Calendar.AUGUST, 14).getTime();
private static int age = 31; //IT WILL CHANGE WITH TIME
````
and the `getAge()` method in `Patient` changes the age depending on the birthdate and the current day;

````
public int getAge() {
   if (this.birthDate != null) {
      GregorianCalendar birthday = new GregorianCalendar();
      birthday.setTime(birthDate);
      DateTime now = new DateTime();
      DateTime birth = new DateTime(birthday.getTime());  
      Period period = new Period(birth, now, PeriodType.yearMonthDay());
      age = period.getYears();
   }
   return age;
}
````

To eliminate that check would take additional changes.

Reply:
@mwithi OK; I created a new method, `setPatientConsistency(Opd opd)`, by moving the setting of age and sex out of the `validateOpd()` routine. The new method is added to `newOpd()`.   This is the same type of change made for `Laboratory`. So, I think this covers all the remaining concerns.

Reply:
Thanks!